docs/src/interfacer.md

```diff
@@ -0,0 +1,56 @@
+# Interfacer
+
+This module contains functions for defining the interface for coupling component models, as well as stub objects that contain prescribed fields.
+
+## Coupled Simulation
+- TODO: move `CoupledSimulation` from Utilities in a separate PR
+- `CoupledSimulation` (`cs`) stores info for ESM run. We require that each `cs` contains an `atmos_sim`, `ocean_sim` and `ice_sim` component. While this requirement will not be eventually needed, for the time being, if a simulation type is not needed, it should be initialized with `SurfaceStub` with a zero `area_fracion`.
+
+## Component model simulations
+- all Simulations that are not the `CoupledSimulation` fall under `ComponentModelSimulation`
+- the current version requires that there is:
+    - one `AtmosModelSimulation`
+    - one or more `SurfaceModelSimulation`s, which require the following adapter functions:
+        ```
+        get_field(sim::SurfaceModelSimulation, ::Val{:area_fraction}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:air_temperature}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:albedo}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:z0m}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:z0b}) = ...
```

Comment:
What are `z0m` and `z0b`? I think we should add this as a comment or docstring since it's not immediately clear from the names

Reply:
These are roughness lengths for momentum and buoyancy. I made these more verbose. We should formalize/unify the fields (and their units) being passed by the coupler though. I'll open an issue on this.